sqlite3.go

```diff
@@ -1123,18 +1123,20 @@ func (s *SQLiteStmt) query(ctx context.Context, args []namedValue) (driver.Rows,
 		done:     make(chan struct{}),
 	}
 
-	go func(db *C.sqlite3) {
-		select {
-		case <-ctx.Done():
+	if ctx.Done() != context.Background().Done() {
```

Comment:
@navytux (I just came across this from the golang/go issue you linked it from). Looking at the stdlib, the correct check is simply `ctx.Done() != nil`. (See [context.go](https://golang.org/src/context/context.go#L243)).

Reply:
```go
if done := ctx.Done(); done != nil {
```

Reply:
@tmthrgd, @mattn, thanks for feedback. Indeed it is even specified in Context documentation that

    Done may return nil if this context can never be canceled.

(https://golang.org/pkg/context/#Context; I was probably under impression of `ctx.Done() == context.Background().Done()` checks I saw in stdlib earlier that were later fixed - e.g. in https://github.com/golang/go/commit/b9dca1b1).

Code fixed to what mattn says.